cpp/src/Slice/Parser.cpp

```diff
@@ -2720,78 +2823,15 @@ Slice::Container::checkIntroduced(const string& scoped, ContainedPtr namedThing)
 }
 
 bool
-Slice::Container::nameIsLegal(const string& newName, const char* newConstruct)
+Slice::Container::checkForGlobalDef(const string& name, const char* newConstruct)
 {
-    ModulePtr module = ModulePtr::dynamicCast(this);
-
-    //
-    // Check whether the enclosing module has the same name.
-    //
-    if(module)
+    if(dynamic_cast<Unit*>(this) && strcmp(newConstruct, "module"))
     {
-        if(newName == module->name())
+        static const string vowels = "aeiou";
+        string glottalStop;
+        if(vowels.find_first_of(newConstruct[0]) != string::npos)
```

Comment:
Looks like this could replaced with your `prependA` function.

There's also a function in the Grammar that should be updated to use your new function:
https://github.com/zeroc-ice/ice/blob/33b7534a599c1b37b3c0fb540a894e8f83287cb8/cpp/src/Slice/Grammar.y#L2120

Reply:
I let you clean this up in a follow-up PR.